module/mempool/herocache/execution_data.go

```diff
@@ -0,0 +1,97 @@
+package herocache
+
+import (
+	"fmt"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/executiondatasync/execution_data"
+	herocache "github.com/onflow/flow-go/module/mempool/herocache/backdata"
+	"github.com/onflow/flow-go/module/mempool/herocache/backdata/heropool"
+	"github.com/onflow/flow-go/module/mempool/herocache/internal"
+	"github.com/onflow/flow-go/module/mempool/stdmap"
+)
+
+type BlockExecutionData struct {
+	c *stdmap.Backend
+}
+
+// NewBlockExecutionData implements a block execution data mempool based on hero cache.
+func NewBlockExecutionData(limit uint32, logger zerolog.Logger, collector module.HeroCacheMetrics) *BlockExecutionData {
+	t := &BlockExecutionData{
+		c: stdmap.NewBackend(
+			stdmap.WithBackData(
+				herocache.NewCache(limit,
+					herocache.DefaultOversizeFactor,
+					heropool.LRUEjection,
+					logger.With().Str("mempool", "block_execution_data").Logger(),
+					collector))),
+	}
+
+	return t
+}
+
+// Has checks whether the block execution data with the given hash is currently in
+// the memory pool.
+func (t BlockExecutionData) Has(id flow.Identifier) bool {
```

Comment:
From the performance perspective, the methods `Has`, `ByID`, `All`, `Size` should have pointer receivers, i.e., `func (t *BlockExecutionData) Has(id flow.Identifier) bool` instead of `func (t BlockExecutionData) Has(id flow.Identifier) bool`. This ensures that _the struct is not copied_ when calling these methods. As `HeroCache` keeps everything by value, copying the entire struct per call can become a performance bottleneck.

Reply:
sounds good. This was copied from `transactions.go`:
https://github.com/onflow/flow-go/blob/1f9eddc257ab10a6804222da3ea72dc43ab2f8e8/module/mempool/herocache/transactions.go#L34-L38

Think we should fix that as well (in a different PR)?

Reply:
[bdaa257](https://github.com/onflow/flow-go/pull/4197/commits/bdaa257c41061d70489910a54f45d61d62fa17b8)